core/src/main/java/org/infinispan/distexec/DefaultExecutorService.java

```diff
@@ -980,10 +980,10 @@ public void execute() {
 
       @Override
       public boolean cancel(boolean mayInterruptIfRunning) {
-         if (!isCancelled()) {
+         if (!isCancelled() && super.cancel(true)) {
```

Comment:
Unfortuantely `super.cancel(true)` will return true even if called the second time.  It is a weird behavior change from regular Future I know :-1: